src/utilities/make_state_vector_file.py

```diff
@@ -138,22 +146,24 @@ def make_state_vector_file(
 
     # Assign buffer pixels (the remaining 0's) to state vector
     # -------------------------------------------------------------------------
-    buffer_area = statevector.values == 0
-
-    # Get image coordinates of all pixels in buffer area
-    irows = np.arange(buffer_area.shape[0])
-    icols = np.arange(buffer_area.shape[1])
-    irows = np.transpose(np.tile(irows, (len(icols), 1)))
-    icols = np.tile(icols, (len(irows), 1))
-    irows_good = irows[buffer_area > 0]
-    icols_good = icols[buffer_area > 0]
-    coords = [[icols_good[j], irows_good[j]] for j in range(len(irows_good))]
-
-    # K-means
-    X = np.array(coords)
-    kmeans = KMeans(n_clusters=k_buffer_clust, random_state=0).fit(X)
+    if is_regional:
+        buffer_area = statevector.values == 0
+
+        # Get image coordinates of all pixels in buffer area
+        irows = np.arange(buffer_area.shape[0])
+        icols = np.arange(buffer_area.shape[1])
+        irows = np.transpose(np.tile(irows, (len(icols), 1)))
+        icols = np.tile(icols, (len(irows), 1))
+        irows_good = irows[buffer_area > 0]
+        icols_good = icols[buffer_area > 0]
+        coords = [[icols_good[j], irows_good[j]] for j in range(len(irows_good))]
+
+        # K-means
+        X = np.array(coords)
+        kmeans = KMeans(n_clusters=k_buffer_clust, random_state=0).fit(X)
 
     # Assign pixels to state vector
+    # -------------------------------------------------------------------------
```

Comment:
kmeans gets calles just below this on line 174 -- shouldn't it be included in the if statement? Since kmeans only handles the buffer cells which are not needed for global?

Reply:
Thanks for catching that! It's now fixed in eee2cc2.